pkg/runner/pretty.go

```diff
@@ -71,13 +71,16 @@ func NewPrettyPrinter() *PrettyPrinter {
 
 // Wait waits for all running tests to finish and returns an error if any of
 // them failed.
-func (c *PrettyPrinter) Wait() error {
-	c.wg.Wait()
-
-	if f := atomic.LoadUint32(&c.failed); f > 0 {
-		return fmt.Errorf("%d nodes failed", f)
-	}
-	return nil
+func (c *PrettyPrinter) Wait() <-chan error {
```

Comment:
This is a weird interface. We seem to only use it at 2 places, so I guess it is fine, but if we start using this PrettyPrinter more often, maybe it'd make sense just introducing 2 methods: `Wait()` as it currently works, and `WaitChan` which returns a channel.